pkg/commands/run_test.go

```diff
@@ -311,6 +312,140 @@ func Test_CachingRunCommand_ExecuteCommand(t *testing.T) {
 	}
 }
 
+func TestFileCreatorCleaner(t *testing.T) {
+	const (
+		fileContent = "content"
+		fileMode    = 0600
+		dirMode     = 0700
+	)
+
+	t.Run("Creator", func(t *testing.T) {
+		dir := t.TempDir()
+		fcc := fileCreatorCleaner{}
+
+		// Given a fileCreatorCleaner
+		// When fileCreatorCleaner.MkdirAndWriteFile is called
+		file := filepath.Join(dir, "file")
+		err := fcc.MkdirAndWriteFile(file, []byte(fileContent), dirMode, fileMode)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		// Then the file should exist
+		if _, err := os.Stat(file); errors.Is(err, os.ErrNotExist) {
```

Comment:
We should really add a linter for these 😄 (not going to comment on the other ones but they should be fixed as well).
```suggestion
		if _, err := filesystem.FS.Stat(file); errors.Is(err, os.ErrNotExist) {
```
(Obviously not critical here, but it's best to stay consistent.)

Reply:
found and fixed remaining occurrences.